src/realm/util/string_buffer.cpp

```diff
@@ -23,21 +23,19 @@
 
 #include <realm/util/safe_int_ops.hpp>
 #include <realm/util/string_buffer.hpp>
+#include <realm/utilities.hpp>
 
 using namespace realm;
 using namespace realm::util;
 
 
-char StringBuffer::m_zero = 0;
-
-
```

Comment:
Its raison d'être seems to be to provide a (canonical?) empty string? This seems to change that behaviour? Maybe just return `""`? That should resolve to the same empty string in the static section of the final binary, no?